examples/spells/rareSpells.qnt

```diff
@@ -9,6 +9,23 @@
  * so the others would see how to use it.
  */
 module rareSpells {
+
+  /// The type of orderings between comparable things
+  type Ordering =
+    | EQ
+    | LT
+    | GT
+
+  /// Comparison of integers
+  pure def intCompare(__a: int, __b:int): Ordering = {
+    if (__a < __b)
+      { LT }
+    else if (__a > __b)
+      { GT }
+    else
+      { EQ }
+  }
```

Comment:
Is it that important to use sum types here. Many languages that I know would use -1, 0, and 1. On the other hand, constant values may be more beneficial for Apalache. So yes, let's keep `LT`, `GT`, and `EQ`. Perhaps, it's worth adding a comment on that.

Reply:
I added a general comment about the motives touching on both things.